app/modules/raid/cruds_raid.py

```diff
@@ -0,0 +1,391 @@
+from datetime import UTC, datetime
+from sqlite3 import IntegrityError
+from typing import Sequence
+
+from sqlalchemy import delete, or_, select, update
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import selectinload
+
+from app.modules.raid import models_raid, schemas_raid
+
+
+async def create_participant(
+    participant: models_raid.Participant,
+    db: AsyncSession,
+) -> models_raid.Participant:
+    db.add(participant)
+    try:
+        await db.commit()
+        return participant
+    except IntegrityError:
+        await db.rollback()
+        raise ValueError("An error occurred while creating the participant.")
+
+
+async def update_participant(
+    participant_id: str,
+    participant: schemas_raid.ParticipantUpdate,
+    db: AsyncSession,
+) -> None:
+    await db.execute(
+        update(models_raid.Participant)
+        .where(models_raid.Participant.id == participant_id)
+        .values(**participant.model_dump(exclude_none=True))
+    )
+    await db.commit()
+
+
+async def is_user_a_participant(
+    user_id: str,
+    db: AsyncSession,
+) -> bool:
+    participant = await db.execute(
+        select(models_raid.Participant).where(models_raid.Participant.id == user_id)
+    )
+    return bool(participant.scalars().first())
+
+
+async def get_team_by_participant_id(
+    participant_id: str,
+    db: AsyncSession,
+) -> models_raid.Team | None:
+    team = await db.execute(
+        select(models_raid.Team)
+        .where(
+            or_(
+                models_raid.Team.captain_id == participant_id,
+                models_raid.Team.second_id == participant_id,
+            )
+        )
+        .options(
+            selectinload("*"),
+        )
+    )
+    return team.scalars().first()
+
+
+async def get_all_teams(
+    db: AsyncSession,
+) -> Sequence[models_raid.Team]:
+    teams = await db.execute(
+        select(models_raid.Team).options(
+            selectinload("*"),
```

Comment:
Could you add a comment explaining `selectinload("*")` allows selecting relationships contained in relationships, like `models_raid.Team.captain.id_card`?